### PR TITLE
is possible to assign subcategory to category

### PR DESCRIPTION
**Pull request by mb83pl**

### CODE REVIEW

File: src/main/java/com/inqoo/trainingservice/app/models/Category.java [hunk 1]
```diff
@@ -57,4 +57,6 @@ public void addSubcategory(Subcategory subcategory) {
     public List<Subcategory> getSubcategoryList() {
         return subcategoryList;
     }
+
```
Comment (adamkulawik) [inline src/main/java/com/inqoo/trainingservice/app/models/Category.java:60]: empty lines

File: src/main/java/com/inqoo/trainingservice/app/models/Subcategory.java [hunk 2]
```diff
@@ -11,10 +11,13 @@
     private String name;
     @Column(length = 300)
     private String description;
+    @ManyToOne(fetch = FetchType.LAZY)
```
Comment (adamkulawik) [inline src/main/java/com/inqoo/trainingservice/app/models/Subcategory.java:14]: maybe we can try without it - but maybe in next step

File: src/main/java/com/inqoo/trainingservice/app/service/CategoryService.java [hunk 3]
```diff
@@ -25,7 +25,7 @@ public CategoryService(CategoryRepository categoryRepository, SubcategoryReposit
     }
 
     public Category saveNewCategory(Category category) {
-        validateInputs(category, category.getName());
+//        validateInputs(category, category.getName());
```
Comment (adamkulawik) [inline src/main/java/com/inqoo/trainingservice/app/service/CategoryService.java:28]: I think we can restore it

File: src/main/java/com/inqoo/trainingservice/app/service/SubcategoryService.java [hunk 4]
```diff
@@ -12,14 +14,22 @@
...
         this.subcategoryRepository = subcategoryRepository;
+        this.categoryRepository = categoryRepository;
     }
 
     public Subcategory saveNewSubcategory(Subcategory subcategory) {
         validateInputs(subcategory, subcategory.getName());
-        return subcategoryRepository.save(subcategory);
+        Optional<Category> category = categoryRepository.findByName(subcategory.getCategory().getName());
+
+        if(category.isPresent()) {
```
Comment (adamkulawik) [inline src/main/java/com/inqoo/trainingservice/app/service/SubcategoryService.java:28]: What if it is not present?

File: src/main/java/com/inqoo/trainingservice/app/models/Category.java (hunk 1 above)
Comment (mb83pl) [reply src/main/java/com/inqoo/trainingservice/app/models/Category.java:60]: Done.

File: src/main/java/com/inqoo/trainingservice/app/models/Subcategory.java (hunk 2 above)
Comment (mb83pl) [reply src/main/java/com/inqoo/trainingservice/app/models/Subcategory.java:14]: Ok.

File: src/main/java/com/inqoo/trainingservice/app/service/CategoryService.java (hunk 3 above)
Comment (mb83pl) [reply src/main/java/com/inqoo/trainingservice/app/service/CategoryService.java:28]: Done.

File: src/main/java/com/inqoo/trainingservice/app/service/SubcategoryService.java (hunk 4 above)
Comment (mb83pl) [reply src/main/java/com/inqoo/trainingservice/app/service/SubcategoryService.java:29]: Done. Added CategoryNotFoundException.

File: src/test/java/com/inqoo/trainingservice/app/service/CategoryServiceIT.java [hunk 5]
```diff
@@ -99,41 +99,37 @@ public void shouldThrowExceptionIfNameCategoryAlreadyTaken() {
...
-
-        //when
-        Category savedCat = categoryService.saveNewCategory(category);
-        Subcategory savedSubcat = subcategoryService.saveNewSubcategory(subcategory);
-        Subcategory savedSubcat2 = subcategoryService.saveNewSubcategory(subcategory2);
-        boolean savedRelation = categoryService.saveNewRelation(savedCat.getId(), savedSubcat.getId());
-        boolean savedRelation2 = categoryService.saveNewRelation(savedCat.getId(), savedSubcat2.getId());
-
-        //then
+//    @Test
```
Comment (adamkulawik) [inline src/test/java/com/inqoo/trainingservice/app/service/CategoryServiceIT.java:102]: this is also commented out

Comment (mb83pl) [reply src/test/java/com/inqoo/trainingservice/app/service/CategoryServiceIT.java]: Done.

**Review by adamkulawik — APPROVED** [review]

**Pull request merged**